test/lib/InputProcessor.test.ts

```diff
@@ -0,0 +1,141 @@
+import {InputProcessor, InputProcessorImpl} from "../../src/lib/InputProcessor";
+import {assert, expect} from "chai";
+import * as path from "path";
+import {Inputs} from "../../src/types";
+import {BundleName, getMessage} from "../../src/MessageCatalog";
+import untildify = require("untildify");
+import normalize = require("normalize-path");
+import {FakeDisplay} from "./FakeDisplay";
+
+describe("InputProcessorImpl Tests", async () => {
+	let display: FakeDisplay;
+	let inputProcessor: InputProcessor;
+	beforeEach(async () => {
+		display = new FakeDisplay();
+		inputProcessor = new InputProcessorImpl("2.11.8", display);
+	});
+
+	describe("resolveTargetPaths Tests", async () => {
+		it("Specified glob target stays as glob", async () => {
+			// Note that we may want to change this behavior in the future instead of waiting to resolve the globs
+			// in the DefaultRuleManager. But for now, adding this test.
+			const inputs: Inputs = {
+				target: ['test\\**\\*.page', '!test/code-fixtures/cpd', '~/*.class']
+			};
+			const resolvedTargetPaths: string[] = inputProcessor.resolveTargetPaths(inputs);
+			expect(resolvedTargetPaths).to.have.length(3);
+			expect(resolvedTargetPaths).to.contain('test/**/*.page');
+			expect(resolvedTargetPaths).to.contain('!test/code-fixtures/cpd');
+			expect(resolvedTargetPaths).to.contain(normalize(untildify('~/*.class')))
+		})
+
+		it("Specified target with method specifier", async () => {
+			const inputs: Inputs = {
+				target: ['test/code-fixtures/apex/SomeTestClass.cls#testMethodWithoutAsserts']
+			};
+			const resolvedTargetPaths: string[] = inputProcessor.resolveTargetPaths(inputs);
+			expect(resolvedTargetPaths).to.have.length(1);
+			expect(resolvedTargetPaths).to.contain('test/code-fixtures/apex/SomeTestClass.cls#testMethodWithoutAsserts');
+		})
+
+		it("Unspecified target resolves to current directory", async () => {
+			const inputs: Inputs = {}
+			const resolvedTargetPaths: string[] = inputProcessor.resolveTargetPaths(inputs);
+			expect(resolvedTargetPaths).to.have.length(1);
+			expect(resolvedTargetPaths).to.contain('.');
+		})
+	})
+
+	describe("resolveProjectDirPath Tests", async () => {
+		it("Specified relative projectdir", async () => {
+			const inputs: Inputs = {
+				projectdir: 'test/code-fixtures'
+			};
+			const resolvedProjectDir: string = inputProcessor.resolveProjectDirPath(inputs);
+			expect(resolvedProjectDir).to.equal(path.resolve('test/code-fixtures'))
+		})
+
+		it("Specified absolute projectdir", async () => {
+			const inputs: Inputs = {
+				projectdir: path.resolve('test/code-fixtures')
```

Comment:
My eyes glazed over the `path.resolve()` here, and I got confused as to how this differed from the relative path test. Can you add a one-line comment drawing attention to that?

Reply:
Isn't path.resolve showing it is an absolute path? Or would a alias subfunction that says toAbsPath help?

Reply:
Yeah, never mind, it's fine as is.